crates/red_knot_workspace/src/site_packages.rs

```diff
@@ -0,0 +1,170 @@
+//! Utilities for finding the `site-packages` directory,
+//! into which third-party packages are installed.
+//!
+//! The routines exposed by this module have different behaviour depending
+//! on the platform of the *host machine*, which may be
+//! different from the *target platform for type checking*. (A user
+//! might be running red-knot on a Windows machine, but might
+//! reasonably ask us to type-check code assuming that the code runs
+//! on Linux.)
+
+use std::io;
+
+use ruff_db::system::{System, SystemPath, SystemPathBuf};
+
+/// Attempt to retrieve the `site-packages` directory
+/// associated with a given Python installation.
+///
+/// `sys_prefix_path` is equivalent to the value of [`sys.prefix`]
+/// at runtime in Python. For the case of a virtual environment, where a
+/// Python binary is at `/.venv/bin/python`, `sys.prefix` is the path to
+/// the virtual environment the Python binary lies inside, i.e. `/.venv`,
+/// and `site-packages` will be at `.venv/Lib/site-packages`. System
+/// Python installations generally work the same way: if a system Python
+/// installation lies at `/opt/homebrew/bin/python`, `sys.prefix` will be
+/// `/opt/homebrew`, and `site-packages` will be at
+/// `/opt/homebrew/Lib/site-packages`.
+///
+/// This routine does not verify that `sys_prefix_path` points
+/// to an existing directory on disk; it is assumed that this has already
+/// been checked.
+///
+/// [`sys.prefix`]: https://docs.python.org/3/library/sys.html#sys.prefix
+#[cfg(target_os = "windows")]
```

Comment:
We generally prefer usinf if cfg over replacing entire functions. The former has the advantage that the code compiles on all platforms. That makes it easier to change the code

Reply:
See https://doc.rust-lang.org/rust-by-example/attribute/cfg.html

Llvm should still remove the dead code

Reply:
Ah, perhaps I misunderstood how `if cfg!()` worked. My understanding was that `if cfg!()` (unlike `#[cfg()]`) didn't remove any code from the shipped binary

Reply:
Sorry, I posted ^that comment after I saw your second comment!

Reply:
I do find the use of the term *runtime* in the `cfg!` documentation confusing. That's why I created a small test in [compiler explorer](https://godbolt.org/z/WEveE9Mc4). Looking at the generated assembly, you can see that only the string "Linux" makes it into the binary. 

The way I understand it is that the Rust compiler only generates code for the matching target when using `#[cfg]`, which is what you want when using platform-specific instructions (because that code wouldn't compile on other platforms). With `cfg!`, rust does generate the instructions for all code but it replaces the `cfg!` expression with a constant `true` or `false` which LLVM then can remove when doing constant folding.

Reply:
I'll fix this up as part of the followup PR I'm working on